src/net/sourceforge/kolmafia/request/MonkeyPawRequest.java

```diff
@@ -0,0 +1,112 @@
+package net.sourceforge.kolmafia.request;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.KoLConstants.MafiaState;
+import net.sourceforge.kolmafia.KoLmafia;
+import net.sourceforge.kolmafia.objectpool.ItemPool;
+import net.sourceforge.kolmafia.preferences.Preferences;
+import net.sourceforge.kolmafia.session.InventoryManager;
+
+public class MonkeyPawRequest extends GenericRequest {
+  private final String wish;
+
+  public MonkeyPawRequest(final String wish) {
+    super("choice.php");
+
+    this.addFormField("whichchoice", "1501");
+    this.addFormField("wish", wish);
+    this.addFormField("option", "1");
+    this.wish = wish.toLowerCase().trim();
+  }
+
+  @Override
+  protected boolean shouldFollowRedirect() {
+    return true;
+  }
+
+  @Override
+  public void run() {
+    if (GenericRequest.abortIfInFightOrChoice()) {
+      return;
+    }
+
+    if (InventoryManager.getAccessibleCount(ItemPool.CURSED_MONKEY_PAW) == 0) {
+      KoLmafia.updateDisplay(MafiaState.ERROR, "You do not have a cursed monkey paw.");
+      return;
+    }
+
+    if (Preferences.getInteger("_monkeyPawWishesUsed") >= 5) {
+      KoLmafia.updateDisplay(MafiaState.ERROR, "You have been cursed enough today.");
+      return;
+    }
+
+    if (!KoLCharacter.hasEquipped(ItemPool.CURSED_MONKEY_PAW)) {
+      InventoryManager.retrieveItem(ItemPool.CURSED_MONKEY_PAW, 1, true);
+    }
+
+    GenericRequest pawRequest = new GenericRequest("main.php?action=cmonk", false);
+    pawRequest.run();
+
+    // Redirects to choice.php?forceoption=0
+    super.run();
+  }
+
+  // You look at your cursed monkey paw.  It has 5 fingers held up expectantly.
+  // You look at your cursed monkey paw.  It has 4 fingers held up expectantly.
+  // You look at your cursed monkey paw.  It has 3 fingers held up expectantly.
+  // You look at your cursed monkey paw.  It has 2 fingers held up expectantly.
+  // You look at your cursed monkey paw.  It has 1 finger held up expectantly.
+  // You look at your cursed monkey paw.  It is closed in a tight withholding fist.
+  private static final Pattern WISH_PATTERN =
+      Pattern.compile("It has (\\d) fingers? held up expectantly\\.");
+
+  public static void visitChoice(final String responseText) {
+    if (responseText.contains("It is closed in a tight withholding fist.")) {
+      Preferences.setInteger("_monkeyPawWishesUsed", 5);
+      return;
+    }
+
+    Matcher matcher = MonkeyPawRequest.WISH_PATTERN.matcher(responseText);
+    if (matcher.find()) {
+      int wishesLeft = Integer.parseInt(matcher.group(1));
+      Preferences.setInteger("_monkeyPawWishesUsed", 5 - wishesLeft);
+    }
+  }
+
+  // You acquire an effect: <b>Cursed by a Monkey</b><br>(duration: 7 Adventures)
+  private static final Pattern CURSE_PATTERN =
+      Pattern.compile("<b>Cursed by a MOnkey</b>.*?(duration: (\\d+) Adventures)");
```

Comment:
```suggestion
      Pattern.compile("<b>Cursed by a Monkey</b>.*?(duration: (\\d+) Adventures)");
```

Perhaps a test for this one?

Reply:
It should actually be tested, although not explicitly:

1) when we visit the monkey paw, we parse the number of fingers and set _monkeyPawWishesUsed
2) when we submit a successful wish, we increment it.
3) and then, we double check by parsing the curse and (we assume) setting the property to the derived value.

If the parsing the visit didn't work, the increment would be inaccurate.
- But the test before I fixed the typo in the pattern gave the expected results.
Now that I fixed the typo, the pattern should match.
- The tests still pass, since the property ends up as expected.

I suppose there could be a test which skips visitChoice and just calls postChoice and verifies it parses the curse effect.

Reply:
Or, perhaps both of my parameterized tests could visit + submit a wish and the first one, which uses GenericRequest could zero out the property before submitting making the wish.

The second one can't be split like that, since it uses an actual MonkeyPawRequest which does both requests. I.e. it tests the automation, not just choice handling.

Reply:
I don't understand how `usingMonkeyPawDetectsWishesUsed` managed to pass while the regex was wrong. It set the preference to 0, then hit the HTML, and then the preference was right -- but looking at the code, it should have just always been 1?

Reply:
Because it parsed the number of fingers in the visit of the intial request.
After submitting the second request to make the wish, it just incremented.